### PR TITLE
ci: breaking change policy

### PR DESCRIPTION
**Pull request by wjones127**

### CODE REVIEW

**Comment by wjones127** [comment]
Tested in https://github.com/wjones127/lance/pull/3

File: release_process.md [hunk 1]
```diff
@@ -56,3 +56,36 @@ want to increment the minor version, such as when there is a breaking change
...
+
+We try to avoid breaking changes, but sometimes they are necessary. When there
+are breaking changes, we will increment the minor version. (This is valid 
+semantic versioning because we are still in `0.x` versions.)
+
+When a PR makes a breaking change, the PR author should mark the PR using the 
+conventional commit markers: either exclamation mark after the type
+(such as `feat!: change signature of func`) or have `BREAKING CHANGE` in the
+body of the PR. A CI job will add a `breaking-change` label to the PR, which is
+what will ultimately be used to CI to determine if the minor version should be
```
Comment (westonpace) [inline release_process.md:71]:
```suggestion
what will ultimately be used by CI to determine if the minor version should be
```

File: release_process.md [hunk 2]
```diff
@@ -56,3 +56,36 @@ want to increment the minor version, such as when there is a breaking change
...
+exception is if it has already been incremented since the last stable release.
+
+**It is the responsibility of the PR author to increment the minor version when
+appropriate.**
+
+Some things that are considered breaking changes:
+
+* Upgrading a dependency pin that is in the Rust API. In particular, upgrading
+  `DataFusion` and `Arrow` are breaking changes. Changing dependencies that are
+  not exposed in our public API are not considered breaking changes.
```
Comment (westonpace) [inline release_process.md:85]: Technically this is only true if we are upgrading from one major version to another.  However, given that has been the case for most releases of arrow and datafusion, I'm not sure it needs to be called out.

File: ci/check_versions.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+"""
```
Comment (westonpace) [inline ci/check_versions.py:1]: This is cool!

**Review by westonpace — APPROVED** [review]

**Review by albertlockett — APPROVED** [review]

File: release_process.md (hunk 2 above)
Comment (wjones127) [reply release_process.md:85]: Depends on our definition of breaking change. For example if we change the pin from `40.0.0` to `40.0.1`, and a user has pinned `=40.0.0`, we will have broken their build. But not really made a meaningful break change behavior wise I think.

**Pull request merged**